Model/Sync/Metadata/Processor.php

```diff
@@ -67,55 +67,51 @@ public function __construct(
                     $this->logger->info(
                         __(
-                            'Updating metadata for Magento Store ID: %1, Name: %2 [START]',
+                            'Skipping Magento Store ID: %1, Name: %2 [Disabled]',
                             $storeId,
                             $this->yotpoConfig->getStoreName($storeId)
                         )
                     );
+                    continue;
+                }
+                $this->logger->info(
+                    __(
+                        'Updating metadata for Magento Store ID: %1, Name: %2 [START]',
+                        $storeId,
+                        $this->yotpoConfig->getStoreName($storeId)
+                    )
+                );
 
-                    $data = $this->prepareMetadata();
-                    $data['entityLog'] = 'general';
-                    $endPoint = $this->yotpoConfig->getEndpoint('metadata');
-                    $response = $this->yotpoSyncMain->syncV1('POST', $endPoint, $data);
-                    if ($response['is_success']) {
-                        $this->logger->info(
-                            __(
-                                'Updating metadata for Magento Store ID: %1, Name: %2 [SUCCESS]',
-                                $storeId,
-                                $this->yotpoConfig->getStoreName($storeId)
-                            )
-                        );
-                    }
-                } catch (\Exception $e) {
+                $data = $this->prepareMetadata();
+                $data['entityLog'] = 'general';
+                $endPoint = $this->yotpoConfig->getEndpoint('metadata');
+                $response = $this->yotpoSyncMain->syncV1('POST', $endPoint, $data);
+                if ($response['is_success']) {
                     $this->logger->info(
                         __(
-                            'Exception on Magento Store ID: %1, Name: %2, Reason: %3',
+                            'Updating metadata for Magento Store ID: %1, Name: %2 [SUCCESS]',
                             $storeId,
-                            $this->yotpoConfig->getStoreName($storeId),
-                            $e->getMessage()
+                            $this->yotpoConfig->getStoreName($storeId)
                         )
                     );
                 }
-                $this->stopEnvironmentEmulation();
+            } catch (\Exception $e) {
+                $this->logger->info(
+                    __(
+                        'Exception on Magento Store ID: %1, Name: %2, Reason: %3',
+                        $storeId,
+                        $this->yotpoConfig->getStoreName($storeId),
+                        $e->getMessage()
+                    )
+                );
             }
-        } catch (\Exception $e) {
-            $this->logger->info(__('Exception: %1', $e->getMessage()));
+            $this->stopEnvironmentEmulation();
```

Comment:
I would check about putting the `stopEnvironmentEmulation` inside a _finally_ clause.

Reply:
added.

Reply:
nice idea, added.